lib/install/validate/validator_test.go

```diff
@@ -175,6 +174,127 @@ func TestMain(t *testing.T) {
 	}
 }
 
+func TestGrantPerms(t *testing.T) {
+	log.SetLevel(log.DebugLevel)
+	trace.Logger.Level = log.DebugLevel
+	ctx := context.Background()
+
+	for i, model := range []*simulator.Model{simulator.ESX(), simulator.VPX()} {
+		t.Logf("%d", i)
+		model.Datastore = 3
+		defer model.Remove()
+		err := model.Create()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		s := model.Service.NewServer()
+		defer s.Close()
+
+		s.URL.User = url.UserPassword("user", "pass")
+		s.URL.Path = ""
+		t.Logf("server URL: %s", s.URL)
+
+		var input *data.Data
+		if i == 0 {
+			input = getESXData(s.URL)
+		} else {
+			input = getVPXData(s.URL)
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		conf := &config.VirtualContainerHostConfigSpec{}
+		opsUser := "ops-user-name"
+		opsPassword := "ops-user-password"
+		opsGrantPermissions := true
+		input.OpsCredentials.OpsUser = &opsUser
+		input.OpsCredentials.OpsPassword = &opsPassword
+
+		// Test 1: current value in conf is the empty string,
+		// input is true, after validation
+		// the new GrantPerms value in conf should be
+		// "config.AddPerms"
+		conf.GrantPermsLevel = ""
+		input.OpsCredentials.GrantPerms = &opsGrantPermissions
+		validator, err := NewValidator(ctx, input)
+		if err != nil {
+			t.Errorf("Failed to new validator: %s", err)
```

Comment:
Failed to create validator.
There is also another testify package, `require`. You could say the following without the `if` clause and it would abort the test immediately. IIRC `t.Errorf` will fail the test but it will still continue to run past that point. I think you'd need `t.Fatalf` which will use `t.FailNow` instead of `t.Fail`:
```
require.NoError(t, err, "Failed to create validator")
```